nemo/collections/tts/inference/pipeline.py

```diff
@@ -0,0 +1,272 @@
+    @abstractmethod
+    def preprocess_text(self, text: str) -> str:
+        """
+        Pre-normalization text processing.
+
+        Args:
+            text: input text string.
+
+        Returns:
+            pre-processed text string.
+        """
+        raise NotImplementedError()
+
+    @abstractmethod
+    def normalize_text(self, text: str) -> str:
+        """
+        Text normalization.
+
+        Args:
+            text: input text string.
+
+        Returns:
+            normalized text string.
+        """
+        raise NotImplementedError()
+
+    @abstractmethod
+    def postprocess_text(self, text: str) -> str:
+        """
+        Post-normalization text processing.
+
+        Args:
+            text: input text string.
+
+        Returns:
+            post-processed text string.
+        """
+        raise NotImplementedError()
+
+
+class GraphemeToPhoneme(ABC):
+    @abstractmethod
+    def convert_graphemes_to_phonemes(self, text: str) -> str:
+        """
+        Convert string containing graphemes/characters to phonemes (eg. IPA or ARPABET).
+          Out of vocabulary words are left as graphemes.
+
+        Args:
+            text: post-processed text string.
+
+        Returns:
+            phoneme string
+        """
+        raise NotImplementedError()
+
+    @abstractmethod
+    def convert_graphemes_to_phonemes_mixed(self, text: str, phone_prob: float) -> str:
+        """
+        Convert each word in input string from graphemes to phonemes with probability phone_prob.
+          Out of vocabulary words are left as graphemes.
+
+        Args:
+            text: post-processed text string.
+
+        Returns:
+            phoneme string
+        """
+        raise NotImplementedError()
+
+
+class TextTokenizer(ABC):
+    @abstractmethod
+    def tokenize(self, text: str) -> List[int]:
+        """
+        Create list of integer indices for the given text string.
+
+        Args:
+            text: input text string.
+
+        Returns:
+            list of integer indices.
+        """
+        raise NotImplementedError()
+
+
+class AudioProcessor(ABC):
+    @abstractmethod
+    def compute_spectrogram(self, audio: torch.tensor, audio_len: torch.tensor) -> Tuple[torch.tensor, torch.tensor]:
+        """
+        Computes the spectrogram of the input audio batch.
+
+        Args:
+            audio: [B, T_audio] float tensor containing [-1, 1] audio samples.
+            audio_len: [B] integer tensor containing length of audio in batch.
+
+        Returns:
+            [B, spec_dim, T_spec] float tensor with spectrogram features.
+            [B] integer tensor with output spectrogram lengths.
+        """
+        raise NotImplementedError()
+
+
+class SpectrogramSynthesizer(ABC):
+    @property
+    @abstractmethod
+    def device(self):
+        pass
+
+    @abstractmethod
+    def synthesize_spectrogram(self, tokens: torch.tensor, speaker: torch.tensor, pitch: torch.tensor) -> torch.tensor:
```

Comment:
That would make sense as it is in our current fastpitch export. Should the pace be a scalar that speeds up or slows down the entire utterance, or the same as pitch with 1 pace value per token? The fastpitch code only supports the former but the latter would make more sense to me.

Also, it would not necessarily be an "option". The functions from tensor to tensor should not have optional arguments so they can be exported. The pipeline class could however handle the logic of creating default inputs for the user (eg. set pitch to duplicated 0.0 and pace to 1.0)